File: 36. Valid Sudoku/36. Valid Sudoku.rb
# @param {Character[][]} board
# @return {Boolean}
def is_valid_sudoku(board)
    rows = Array.new(9) { Array.new(9, false) }
    cols = Array.new(9) { Array.new(9, false) }
    boxes = Array.new(9) { Array.new(9, false) }
    
    9.times do |i|
        9.times do |j|
            if board[i][j] != '.'
                num = board[i][j].to_i - 1
                box_index = (i / 3) * 3 + (j / 3)
                
                return false if rows[i][num] || cols[j][num] || boxes[box_index][num]
                
                rows[i][num] = cols[j][num] = boxes[box_index][num] = true
            end
        end
    end
    
    true
end

File: 36. Valid Sudoku/36. Valid Sudoku.c
bool isValidSudoku(char** board, int boardSize, int* boardColSize) {
    int rows[9][9] = {0};
    int cols[9][9] = {0};
    int boxes[9][9] = {0};
    
    for (int i = 0; i < 9; i++) {
        for (int j = 0; j < 9; j++) {
            if (board[i][j] != '.') {
                int num = board[i][j] - '1';
                int box_index = (i / 3) * 3 + j / 3;
                
                if (rows[i][num] || cols[j][num] || boxes[box_index][num]) {
                    return false;
                }
                
                rows[i][num] = cols[j][num] = boxes[box_index][num] = 1;
            }
        }
    }
    
    return true;
}